src/data/automation_i18n.ts

```diff
@@ -400,9 +400,25 @@ const tryDescribeTrigger = (
       return `${entityStr}${offsetStr}`;
     });
 
-    return hass.localize(`${triggerTranslationBaseKey}.time.description.full`, {
-      time: formatListWithOrs(hass.locale, result),
-    });
+    // Handle weekday information if present
+    let weekdayStr = "";
+    if (trigger.weekday) {
+      const weekdayArray = ensureArray(trigger.weekday);
+      if (weekdayArray.length > 0) {
+        const localizedDays = weekdayArray.map((day) =>
+          hass.localize(
+            `ui.panel.config.automation.editor.triggers.type.time.weekdays.${day}` as any
+          )
+        );
+        weekdayStr = ` on ${formatListWithOrs(hass.locale, localizedDays)}`;
```

Comment:
This needs to be added to the localized string so it can be properly translated. 

`When the time is equal to {time}{hasWeekdays, select, \n true { on {weekdays}} \n other {}\n }` 

```
return (
      hass.localize(`${triggerTranslationBaseKey}.time.description.full`, {
        time: formatListWithOrs(hass.locale, result),
        hasWeekdays: weekdays.length > 0,
        weekdays: formatListWithOrs(hass.locale, weekdays)
      })
    );
```
where weekdayStr is replaced with weekdays as array